qiita_ware/private_plugin.py

```diff
@@ -303,19 +303,24 @@ def delete_study(job):
         study_id = job.parameters.values['study']
         study = qdb.study.Study(study_id)
 
-        for a in study.analyses():
-            artifacts = sorted(
-                a.artifacts, key=lambda a: a.id, reverse=True)
-            for artifact in artifacts:
-                qdb.artifact.Artifact.delete(artifact.id)
-            qdb.analysis.Analysis.delete(a.id)
-
-        artifacts = sorted(
-            study.artifacts(), key=lambda a: a.id, reverse=True)
-        for a in artifacts:
-            qdb.artifact.Artifact.delete(a.id)
+        # deleting analyses
+        for analysis in study.analyses():
+            # selecting roots of the analysis, can be multiple
+            artifacts = [a for a in analysis.artifacts
+                         if a.processing_parameters is None]
+            # deleting each of the processing graphs
+            for a in artifacts:
+                to_delete = list(a.descendants.nodes())
```

Comment:
confirmed this appears to be recursive. (See CREATE FUNCTION qiita.artifact_descendants)